app/src/main/java/com/mstudio/superheroarch/RickAndMortyRepository.kt

```diff
@@ -1,43 +1,55 @@
 package com.mstudio.superheroarch
 
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
+import java.io.IOException
 
-class RickAndMortyRepository(private val apiRick: ApiRick) {
+class RickAndMortyRepository(private val apiRick: ApiRick, private val characterDao: CharacterDao) {
```

Comment:
we talked about this. Here you need to use the source of the data, from remote you are using the Api, from local you are using the DB, you shouldn't use the Dao. Keep the consistency.

Reply:
Fixed in next commit.